### ISSUE TITLE
A dependency on spring-boot-actuator stops @EnableWebSecurity from switching off the default web security configuration completely

### ISSUE THREAD

**Issue by gregturn** [spring-projects/spring-boot#6447]
Bug Report:

Spring Boot - 1.3.6.RELEASE
Spring Security - 4.1.1.RELEASE

When you have both Spring Boot Actuator and Spring Security, UriTemplate-based routes that attempt to use `SecurityContextHolder.getContext().getAuthentication()` (or use `@AuthenticationPrincipal` in the method signature) will get a `null` value.

Drop Actuator, and the problem vanishes.

For reproducible repository => https://github.com/gregturn/spring-boot-actuator-and-spring-security
To reproduce, follow steps listed => https://github.com/gregturn/spring-boot-actuator-and-spring-security/blob/master/src/main/resources/templates/index.html

**Comment by gregturn** [comment]
FYI: spring-boot-starter-remote-shell pulls in Actuator, so that will also trigger this issue.

**Comment by wilkinsona** [comment]
The problem seems to occur with Spring Security 4.0.x as well

**Comment by wilkinsona** [comment]
The difference in behaviour is because the request mapping's are different. There are three mappings in question:

 1. `@RequestMapping(value = "/")`
 2. `@RequestMapping(method = RequestMethod.GET, value = "/images/{filename:.+}")`
 3. `@RequestMapping(method = RequestMethod.GET, value = "/image")`

It's the second of these that is passed a null authentication principal. That's the expected behaviour as Boot's default ignored paths are `/css/**`, `/js/**`, `/images/**`, and `/**/favicon.ico` and the mapping's path is `/images/{filename:.+}`.

In short the behaviour when Actuator on the classpath is correct. I don't yet understand why the behaviour changes when Actuator is not on the classpath.

**Comment by wilkinsona** [comment]
`SecurityConfiguration` in the sample app is annotated with `@EnableWebSecurity` which, [as described in the documentation](http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-security), is used "to switch off the default web security configuration completely".

The actual bug here is that when Actuator is on the classpath, using `@EnableWebSecurity` doesn't switch off the default web security configuration completely.

**Comment by philwebb** [comment]
@wilkinsona Something we need to target to 1.4? I can have a look today if you've got any ideas?

**Comment by wilkinsona** [comment]
@philwebb Yeah, probably. I think the fix is a one-liner: add `@ConditionalOnMissingBean(WebSecurityConfiguration.class)` to `ManagementWebSecurityAutoConfiguration`. There's also the inner `WebSecurityEnabler` class that I think would become redundant. However, we'd need to be careful that we didn't end up in the situation that `FallbackWebSecurityAutoConfiguration` seeks to avoid. The management server being on a separate port probably complicates things further.

TL;DR 2 minutes to add a line and remove `WebSecurityEnabler` and quite a lot longer to make sure everything still works :(

**Comment by philwebb** [comment]
@wilkinsona I'm afraid #6451 ate all my time so I didn't get to this.

**Comment by wilkinsona** [comment]
My proposed fix doesn't work. `SpringBootWebSecurityConfiguration` is annotated with `@EnableWebSecurity` so it creates a `WebSecurityConfiguration` bean. If `ManagementWebSecurityAutoConfiguration` is annotated with `@ConditionalOnMissingBean(WebSecurityConfiguration.class)` then management security is always going to be turned off.

Unfortunately, I can't see an easy way to fix this without some fairly major surgery. It's too late for that in 1.4. AFAICT, the problem described in this issue has existed for quite some time, perhaps even since 1.0. I think we should revisit this early in 1.5 or 2.0.

**Comment by alidufour** [comment]
Maybe this is of help, I am no Spring Guru, but I seem to have the same issue pattern when I throw an Authentication Exception in my custom Authentication Provider (Principal cannot be null), but only happens when Actuator is in project
Happens since migrated to 1.4, and problem doesn't occur when I remove Actuator 
see here for explanation:
[(http://stackoverflow.com/questions/38958004/spring-boot-1-4-principal-must-not-be-null-exception)]

**Comment by mbhave** [comment]
similar to #6137
